pkg/channelserver/channelserver.go

```diff
@@ -171,22 +170,25 @@ func getDefaultFromAppDefaultsByRuntimeAndServerVersion(ctx context.Context, run
 		return "", fmt.Errorf("faild to parse %s defaultVersionRange for %s: %v", runtime, serverVersion, err)
 	}
 
-	var candidate []string
+	var candidate []semver.Version
 	for _, release := range config.ReleasesConfig().Releases {
 		version, err := semver.ParseTolerant(release.Version)
 		if err != nil {
 			logrus.Debugf("fails to parse the release version %s: %v", release.Version, err)
 			continue
 		}
 		if dvrParsed(version) {
-			candidate = append(candidate, release.Version)
+			candidate = append(candidate, version)
 		}
 	}
 	if len(candidate) == 0 {
 		return "", fmt.Errorf("no %s version is found for %s", runtime, serverVersion)
 	}
-	sort.Strings(candidate)
-	return candidate[len(candidate)-1], nil
+	// the build metadata parts are ignored when sorting versions,
+	// thankfully KDM does not contain two versions that differ only in the build metadata,
+	// so we do not need to worry about the order for such case.
```

Comment:
Well, we actually may with delay of Feb 2023 patches by upstream there is a high chance there will be `v1.24.10+rke2r2` released that we will add in a separate release alongside `v1.24.10+rke2r1` :(

Still, let's proceed forward to get this fix done and then do the follow up. Please update the comment to clarify:

```suggestion
	// the build metadata parts are ignored when sorting versions for now;
	// ideally they should be since k3s/RKE2 releases use it to establish order of precedence
```

Reply:
fixed.